shell/platform/darwin/macos/framework/Source/FlutterViewController.mm

```diff
@@ -526,14 +537,24 @@ - (NSDictionary*)getClipboardData:(NSString*)format {
 }
 
 - (void)setClipboardData:(NSDictionary*)data {
-  NSPasteboard* pasteboard = [NSPasteboard generalPasteboard];
+  NSPasteboard* pasteboard = [self _pasteboard];
   NSString* text = data[@"text"];
+  [pasteboard clearContents];
   if (text && ![text isEqual:[NSNull null]]) {
-    [pasteboard clearContents];
     [pasteboard setString:text forType:NSPasteboardTypeString];
   }
 }
 
+- (BOOL)clipboardHasStrings {
+  NSDictionary* data = [self getClipboardData:[NSString stringWithFormat:@"%s", kTextPlainFormat]];
+  NSString* string = data[@"text"];
```

Comment:
Looking at this more closely this time: going through a method whose input and output are explicitly designed for the Flutter method channel contract seems like a very convoluted way to check this. Why not just `return [self.pasteboard stringForType:NSPasteboardTypeString]].length > 0;`?

Reply:
Good point, done.